dtfs-central-api/src/helpers/get-fee-record-payment-entity-group-reconciliation-data.ts

```diff
@@ -0,0 +1,59 @@
+import { orderBy } from 'lodash';
+import { FEE_RECORD_STATUS } from '@ukef/dtfs2-common';
+import { TfmUsersRepo } from '../repositories/tfm-users-repo';
+import { FeeRecordReconciledByUser } from '../types/utilisation-reports';
+import { FeeRecordPaymentEntityGroup } from '../types/fee-record-payment-entity-group';
+
+/**
+ * Gets the reconciliation data for a fee record payment entity group
+ * @param group - The fee record payment entity group
+ * @returns The group reconciliation data
+ * @throws {Error} If the groups fee records array is empty
+ * @throws {Error} If any of the fee records at the `RECONCILED` status have a null `dateReconciled` property
+ */
+export const getFeeRecordPaymentEntityGroupReconciliationData = async (
+  group: FeeRecordPaymentEntityGroup,
+): Promise<{
+  reconciledByUser?: FeeRecordReconciledByUser;
+  dateReconciled?: Date;
+}> => {
+  if (group.feeRecords.length === 0) {
+    throw new Error('Fee record payment entity group cannot have an empty fee records array');
+  }
+
+  const reconciledFeeRecords = group.feeRecords.filter(({ status }) => status === FEE_RECORD_STATUS.RECONCILED);
+
+  if (reconciledFeeRecords.length === 0) {
+    return {};
+  }
+
+  if (reconciledFeeRecords.some(({ dateReconciled }) => dateReconciled === null)) {
```

Comment:
could this be moved to a const like line 24 please? easier to read

Reply:
I don't think it would be easier to read for a simple check like this and instead just makes the code more verbose

Reply:
I agree @Zainzzkk , the current if statement takes more effort to read. I think any conditional logic should have a `const` so it's quicker/easier to understand what it's checking. Something like this:

```ts
const allRecordsHaveReconciledDate = reconciledFeeRecords.some(({ dateReconciled }) => dateReconciled);

if (allRecordsHaveReconciledDate) {
  // ...
}

Reply:
File has changed a lot since this comment - this condition is no longer relevant. Resolving